mithril-test-lab/mithril-devnet/mkfiles/mkfiles-init.sh

```diff
@@ -65,20 +65,21 @@ if ! mkdir -p "${ARTIFACTS_DIR}"; then
   exit
 fi
 
+echo ">> Creating artifacts directory..."
 # Download cardano-cli & cardano-node if enabled (default: yes)
 if [[ "$SKIP_CARDANO_BIN_DOWNLOAD" != "true" ]]; then
   echo ">> Downloading cardano-cli & cardano-node..."
-  curl -sL ${CARDANO_BINARY_URL} --output cardano-bin.tar.gz
-  tar xzf cardano-bin.tar.gz ./bin/cardano-cli ./bin/cardano-node && mv ./bin/cardano-{cli,node} . && rm -rf ./bin  || tar xzf cardano-bin.tar.gz ./cardano-cli ./cardano-node
+  curl -sL "${CARDANO_BINARY_URL}" --output cardano-bin.tar.gz
+  echo ">> Extracting cardano-cli & cardano-node..."
+  tar xzf cardano-bin.tar.gz ./bin || (mkdir -p ./bin && tar --strip-components=1 -C ./bin -xzf cardano-bin.tar.gz)
   rm -f cardano-bin.tar.gz
```

Comment:
You could directly decompress to the artifact directory avoiding the cp/rm that come afterward :).
```suggestion
  tar xzf cardano-bin.tar.gz ./bin -C "${ARTIFACTS_DIR}/" || (mkdir -p "${ARTIFACTS_DIR}/bin" && tar --strip-components=1 -C "${ARTIFACTS_DIR}/bin" -xzf cardano-bin.tar.gz)
  rm -f cardano-bin.tar.gz
```

Reply:
thank you @Alenar